From eda86237c1c3dd4f1cac82b2e164698f16d879dd Mon Sep 17 00:00:00 2001
From: Olivia Crain <oliviacrain@microsoft.com>
Date: Tue, 21 Jun 2022 16:33:24 +0000
Subject: [PATCH] common/reg.c: Fix format-security warnings

Version 6.3 of ncurses introduced function attributes that cause
GCC to recognize more instances of format-security warnings.

In this case, `mvwprintw` is used with strings that have no format
arguments. These strings are effectively handed off to `printf`,
which triggers the format-security warnings.

The proper function to use here is `mvwaddstr`, which serves the
same purpose without passing the string through `printf`.
---
 common/reg.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/common/reg.c b/common/reg.c
index 1a87161..8a66706 100644
--- a/common/reg.c
+++ b/common/reg.c
@@ -240,7 +240,7 @@ reg_line_write(win_reg_t *r, int line, reg_align_t align, char *content)
 	}
 
 	if (len > 0) {
-		(void) mvwprintw(r->hdl, line, pos_x, content);
+		(void) mvwaddstr(r->hdl, line, pos_x, content);
 	}
 
 	if (r->mode != 0) {
@@ -267,7 +267,7 @@ reg_highlight_write(win_reg_t *r, int line, int align, char *content)
 	}
 
 	if (len > 0) {
-		(void) mvwprintw(r->hdl, line, pos_x, content);
+		(void) mvwaddstr(r->hdl, line, pos_x, content);
 	}
 
 	(void) wattroff(r->hdl, A_REVERSE | A_BOLD);
@@ -420,9 +420,9 @@ reg_curses_init(boolean_t first_load)
 
 	if ((g_scr_height < 24 || g_scr_width < 80)) {
 		if (!first_load) {
-			(void) mvwprintw(stdscr, 0, 0,
+			(void) mvwaddstr(stdscr, 0, 0,
 			    "Terminal size is too small.");
-			(void) mvwprintw(stdscr, 1, 0,
+			(void) mvwaddstr(stdscr, 1, 0,
 			    "Please resize it to 80x24 or larger.");
 			(void) refresh();
 		} else {
-- 
2.34.1

